src/bitcoin-chainstate.cpp

```diff
@@ -119,14 +123,12 @@ int main(int argc, char* argv[])
         .chainparams = chainman_opts.chainparams,
         .blocks_dir = abs_datadir / "blocks",
         .notifications = chainman_opts.notifications,
+        .block_tree_db_dir = abs_datadir / "blocks" / "index",
+        .block_tree_db_cache_size = cache_sizes.block_tree_db,
```

Comment:
In commit "kernel: Move block tree db open to block manager" (182945e946ac6a92daad012d6579e5441b9641cc)

It seems far from ideal for bitcoin-chainstate and other kernel library users (not to mention our own test code) to need to hardcode the "blocks/index" path and to manually copy the cache size from one struct to another. I think a better API would allow these option fields to be unset and internally use a `Flatten` method similar to the ones in `validation.cpp` and `txmempool.cpp` to set appropriate default values when the fields are unset.

Reply:
I like the suggestion, but after implementing it still does not seem entirely ideal. I just made the entire `DBParams` field an optional, so it's not easily possible to set only the cache size, but not the path. It is also annoying that the current behaviour with the path is that it is dependent on the data dir path, not the blocks dir path.